src/rpdk/core/templates/hook-docs-readme.md

```diff
@@ -0,0 +1,35 @@
+{% autoescape false %}
```

Comment:
Can you provide an example of how the Generated README would look?

Reply:
# AWS::FooBar::Hook

## Activation

To activate a hook in your account, use the following JSON as the `Configuration` request parameter for [`SetTypeConfiguration`](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_SetTypeConfiguration.html) API request.

### Configuration

<pre>
{
    "CloudFormationConfiguration": {
        "<a href="https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks-structure.html#hooks-hook-configuration" title="HookConfiguration">HookConfiguration</a>": {
            "<a href="https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks-structure.html#hooks-targetstacks" title="TargetStacks">TargetStacks</a>": "ALL",
            "<a href="https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks-structure.html#hooks-failuremode" title="FailureMode">FailureMode</a>": "FAIL",
            "<a href="https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks-structure.html#hooks-properties" title="Properties">Properties</a>" : {
                "<a href="#encryptionalgorithm" title="EncryptionAlgorithm">EncryptionAlgorithm</a>" : <i>String</i>
            }
        }
    }
}
</pre>

Reply:
Can you also include the `Properties` section bellow that describes the actual properties? Similar to [how it is done with resources](https://github.com/aws-cloudformation/cloudformation-cli/blob/master/src/rpdk/core/templates/docs-readme.md?plain=1#L45-L86). 

Don't need the resource specific info (e.g `Update requires`) but its nice to have the properties listed out with the property type, description for the property, and any allowed values (if applicable) which are generated from the schema.

Reply:
Also for the `TargetStacks` and `FailureMode`, since those are controlled by CFN can you update those fields to show the possible values instead. Like `TargetStacks`: "ALL" | "NONE". Similar to how enum values are documented [here](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation.html#CloudFormation.Client.activate_type)

Reply:
# AWS::FooBar::Hook

## Activation

To activate a hook in your account, use the following JSON as the `Configuration` request parameter for [`SetTypeConfiguration`](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_SetTypeConfiguration.html) API request.

### Configuration

<pre>
{
    "CloudFormationConfiguration": {
        "<a href="https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks-structure.html#hooks-hook-configuration" title="HookConfiguration">HookConfiguration</a>": {
            "<a href="https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks-structure.html#hooks-targetstacks" title="TargetStacks">TargetStacks</a>": "ALL" | "NONE",
            "<a href="https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks-structure.html#hooks-failuremode" title="FailureMode">FailureMode</a>": "WARN" | "FAIL",
            "<a href="https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks-structure.html#hooks-properties" title="Properties">Properties</a>" : {
                "<a href="#enablebackup" title="EnableBackup">EnableBackup</a>" : <i>Boolean</i>,
                "<a href="#backupbucketproperties" title="BackupBucketProperties">BackupBucketProperties</a>" : <i>String</i>,
                "<a href="#encryption" title="Encryption">Encryption</a>" : <i>String</i>
            }
        }
    }
}
</pre>


## Properties

#### EnableBackup

_Required_: No

_Type_: Boolean


#### BackupBucketProperties

_Required_: No

_Type_: String


#### Encryption

Required encryption algorithm for hook

_Required_: Yes

_Type_: String

_Allowed Values_: <code>AES256</code> | <code>aws:kms</code>

Reply:
Something like the above. See how the properties are listed with their type, if they are required, and a the description from the schema for that property as well as any allowed values if its an enum

Reply:
Added in new revision